llvm/utils/TableGen/Common/GlobalISel/GlobalISelMatchTable.cpp

```diff
@@ -1726,9 +1738,10 @@ OperandMatcher &InstructionMatcher::addPhysRegInput(Record *Reg, unsigned OpIdx,
 
 void InstructionMatcher::emitPredicateOpcodes(MatchTable &Table,
                                               RuleMatcher &Rule) {
-  if (NumOperandsCheck)
-    InstructionNumOperandsMatcher(InsnVarID, getNumOperands())
+  if (canAddNumOperandsCheck()) {
```

Comment:
Are the curly braces necessary for semantics?

Reply:
I think the coding style recommends to use curly braces if the body of the if is more than 1 line, even if it's just a single statement that is more than 80 cols.

Reply:
No!
https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements

Reply:
"Similarly, braces should be used when a single-statement body is complex enough that it becomes difficult to see where the block containing the following statement began. An if/else chain or a loop is considered a single statement for this rule, and this rule applies recursively."

I will always vote in favor of braces

Reply:
I think it's just a grey area. I'm the primary contributor to these files and I always use curly braces when the if's body spans multiple lines, and I think consistency takes priority.